docs/docs-beta/docs/guides/external-systems/non-python.md

```diff
@@ -1,4 +1,66 @@
 ---
 title: "Running non-Python languages"
 sidebar_position: 60
----
\ No newline at end of file
+---
+
+Dagster is written in Python, but that doesn't mean it's limited to running Python to materialize assets. With Pipes, you can run code in other languages and send information back forth with Dagster. This guide covers how to run JavaScript with Dagster using Pipes, however, the same principle will apply to other languages.
+
+<details>
+<summary>Prerequisites</summary>
+
+- Familiarity with [Assets](/concepts/assets)
+- A basic understanding of JavaScript and Node.js
+</details>
+
+## Create a script using Tensorflow in JavaScript
+
+In this example, we will orchestrate a JavaScript script that reads in a CSV file, and uses the Tensorflow  to train a sequential model using Dagster Pipes.
+
+Start by creating a file named `tensorflow/main.js` with the following contents:
+
+<CodeExample filePath="guides/non-python/pipes-contrived-javascript.js" language="javascript" title="tensorflow/main.js" />
+
+## Create a Dagster asset that runs your script
+
+Create an asset that takes the `PipesSubprocessClient` resource, and set the `compute_kind` so that we know an alternate compute is being used for materialization. Then, use the resource to run your script using `node`.
+
+:::note
+You will need `node` in the environment you are running Dagster, along with Tensorflow installed.
+:::
```

Comment:
note: this might actually be the most annoying step -- to figure out how to correctly install node, esp in envs like docker.

just a note for followup later.

Reply:
added todo